source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -277,9 +283,56 @@ namespace PlayFab
 
                                     } while (dwSize > 0);
 
+                                    std::string requestId;
                                     if (bResults)
                                     {
-                                        ProcessResponse(reqContainer, dwStatusCode);
+                                        constexpr wchar_t requestIdHeaderName[] = L"X-RequestId";
+
+                                        DWORD wideRequestIdByteCount;
+                                        bResults = WinHttpQueryHeaders(
+                                            hRequest,
+                                            WINHTTP_QUERY_CUSTOM,
+                                            requestIdHeaderName,
+                                            WINHTTP_NO_OUTPUT_BUFFER,
+                                            &wideRequestIdByteCount,
+                                            WINHTTP_NO_HEADER_INDEX);
+
+                                        std::unique_ptr<wchar_t[]> wideRequestIdBuffer;
+                                        DWORD queryError = GetLastError();
+                                        if (queryError == ERROR_INSUFFICIENT_BUFFER)
+                                        {
+                                            // WinHttpQueryHeaders excludes the null terminator from the length
+                                            DWORD wideRequestIdLength = wideRequestIdByteCount / sizeof(wideRequestIdBuffer[0]);
+                                            wideRequestIdBuffer = std::make_unique<wchar_t[]>(wideRequestIdLength + 1);
```

Comment:
>std::make_unique [](start = 66, length = 16)

This is available only in C++ 14. I thought we want to support C++ 11? If yes, please consider replacing it with a new equivalent